### PR TITLE
Add com.holypangolin.Animatch

### PR DESCRIPTION
**Pull request by dos1**
Animatch: a mobile-friendly match-three game with cute animals.

https://gitlab.com/HolyPangolin/animatch

### Please confirm your submission meets all the criteria

- [X] Please describe your application briefly.
- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an author/developer/upstream contributor of the project.
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
  - improved AppStream metainfo will be included in the next upstream release
  - Allegro patch is a workaround for a bug in SDL version used by the runtime, but can't be used upstream in this form as it would break other applications

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

**Comment by dos1** [comment]
bot, build com.holypangolin.Animatch

File: com.holypangolin.Animatch.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    <color type="primary" scheme_preference="light">#a6b942</color>
+    <color type="primary" scheme_preference="dark">#838f27</color>
+  </branding>
+
+  <url type="homepage">https://holypangolin.com</url>
+  <url type="donation">https://ko-fi.com/holypangolin</url>
+  <url type="vcs-browser">https://gitlab.com/HolyPangolin/animatch</url>
+  <url type="bugtracker">https://gitlab.com/HolyPangolin/animatch/-/issues</url>
+
+  <content_rating type="oars-1.0" />
```
Comment (hfiguiere) [inline com.holypangolin.Animatch.appdata.xml:64]:
```suggestion
  <content_rating type="oars-1.1" />
```

File: com.holypangolin.Animatch.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        path: stop-text-input.patch
+  - name: animatch
+    buildsystem: cmake-ninja
+    builddir: true
+    sources:
+      - type: archive
+        url: https://holypangolin.com/stuff/animatch/animatch-1.0.3.tar.gz
+        sha256: cea45539385007e240960d30694b0499778718d515eeb71bd23fe6f560db5398
+      - type: file
+        path: com.holypangolin.Animatch.appdata.xml
```
Comment (hfiguiere) [inline com.holypangolin.Animatch.yml:46]: this is never used.

Comment (dos1) [reply com.holypangolin.Animatch.yml:46]: It overwrites the outdated file in the source package and then gets installed as usual.

Comment (hfiguiere) [reply com.holypangolin.Animatch.yml:46]: then it should be a patch.

**Comment by dos1** [comment]
bot, build com.holypangolin.Animatch

**Comment by dos1** [comment]
> ## appstream-missing-screenshots
>
> No image tag of screenshots in the [AppStream catalog](https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#checking-the-generated-output) file has [type](https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-screenshots) thumbnail. Thumbnail images are generated automatically by appstreamcli compose when the image in the [MetaInfo file](https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#path-and-filename) meets the required size.
>
> This can happen when screenshot tags are missing/not properly defined in the [MetaInfo file](https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#path-and-filename) or if the image file/URL is malformed or is below the required size.
>
> Please see the [MetaInfo guidelines](https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#screenshots) on how to define them.

Seems like there's some undocumented image size threshold that needs to be passed. [MetaInfo guidelines](https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#screenshots) don't define it. The closest I could find was this mention in [Quality guidelines](https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/quality-guidelines#screenshots):

> ## Reasonable window size
> In order to ensure that text and interface elements are visible scaled-down in the app store UI the window size should be 1000x700 pixels or smaller (2000x1400 for HiDPI).

...but that's part of guidelines that "are not required for submission to Flathub, but are best practices we recommend and consider for curation and promotion", and don't really apply in this case as it's a fullscreen game and other parts of the guidelines require such fullscreen screenshots in cases like this.

Should I just scale the images down manually? They're currently 720x1440, the native resolution of the game. It's letterboxed on screens with different proportions, but I don't think including big black bars on the sides would improve the quality of screenshots there.

**Comment by bbhtt** [comment]
The issue is appstream is not generating the thumbnails here, likely due to mobile nature of the screenshots.

You include at least one image with width/height decreased so that it can generate the thumbnail. Something like this will work https://x0.at/Jq51.png

**Comment by dos1** [comment]
bot, build com.holypangolin.Animatch

**Comment by dos1** [comment]
After removing `scale="2"` (see https://github.com/ximion/appstream/blob/main/compose/asc-utils-screenshots.c#L603-L605), `appstreamcli compose`  started generating thumbnails for original screenshots properly:
```xml
<screenshots>
  <screenshot>
    <caption>Main menu</caption>
    <image type="source" width="720" height="1440">com/holypangolin/Animatch/f943bbbc013ff9dc36dbc0d8326f55d1/screenshots/image-1_orig.png</image>
    <image type="thumbnail" width="624" height="1248">com/holypangolin/Animatch/f943bbbc013ff9dc36dbc0d8326f55d1/screenshots/image-1_624x1248@1.png</image>
    <image type="thumbnail" width="224" height="448">com/holypangolin/Animatch/f943bbbc013ff9dc36dbc0d8326f55d1/screenshots/image-1_224x448@1.png</image>
  </screenshot>
  <screenshot type="default">
    <caption>Challenge mode</caption>
    <image type="source" width="720" height="1440">com/holypangolin/Animatch/f943bbbc013ff9dc36dbc0d8326f55d1/screenshots/image-2_orig.png</image>
    <image type="thumbnail" width="624" height="1248">com/holypangolin/Animatch/f943bbbc013ff9dc36dbc0d8326f55d1/screenshots/image-2_624x1248@1.png</image>
    <image type="thumbnail" width="224" height="448">com/holypangolin/Animatch/f943bbbc013ff9dc36dbc0d8326f55d1/screenshots/image-2_224x448@1.png</image>
  </screenshot>
  <screenshot>
    <caption>Endless mode</caption>
    <image type="source" width="720" height="1440">com/holypangolin/Animatch/f943bbbc013ff9dc36dbc0d8326f55d1/screenshots/image-3_orig.png</image>
    <image type="thumbnail" width="624" height="1248">com/holypangolin/Animatch/f943bbbc013ff9dc36dbc0d8326f55d1/screenshots/image-3_624x1248@1.png</image>
    <image type="thumbnail" width="224" height="448">com/holypangolin/Animatch/f943bbbc013ff9dc36dbc0d8326f55d1/screenshots/image-3_224x448@1.png</image>
  </screenshot>
</screenshots>
```
flatpak-builder-lint is still complaining about `appstream-missing-screenshots` on my local repo created with flatpak-builder:
```xml
dos@pliszka:~/git/flathub$ ostree cat --repo=repo app/com.holypangolin.Animatch/x86_64/master /files/share/app-info/xmls/com.holypangolin.Animatch.xml.gz | gunzip - | grep screenshot -A 5 -B 5
      <category>LogicGame</category>
    </categories>
    <provides>
      <binary>animatch</binary>
    </provides>
    <screenshots>
      <screenshot>
        <caption>Main menu</caption>
        <image type="source" width="720" height="1440">https://holypangolin.com/stuff/animatch/screenshots/1.png</image>
      </screenshot>
      <screenshot type="default">
        <caption>Challenge mode</caption>
        <image type="source" width="720" height="1440">https://holypangolin.com/stuff/animatch/screenshots/2.png</image>
      </screenshot>
      <screenshot>
        <caption>Endless mode</caption>
        <image type="source" width="720" height="1440">https://holypangolin.com/stuff/animatch/screenshots/3.png</image>
      </screenshot>
    </screenshots>
    <branding>
      <color type="primary" scheme_preference="light">#a6b942</color>
      <color type="primary" scheme_preference="dark">#838f27</color>
    </branding>
    <releases>
```
...but apparently that doesn't match what Flathub is doing at all, since the build has passed now ¯\\_(ツ)_/¯

**Pull request closed without merging**